### PR TITLE
[need-docs][processing] merge grid algorithms and add point grid type

### PR DESCRIPTION
**Pull request by nirvn**
## Description
@nyalldawson , one more processing cleanup we should do prior to release of 3.0 (when it'll be much harder / impossible to remove and rename algorithms): merge our two grid algorithms into one. 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nirvn** [comment]
@3nids , BTW, while working on this, I noticed the QgsLineString constructor that takes a list of QgsPointXY isn't working in python. I'm assuming it's because we already define a constructor with a list of QgsPoint. Should we add a SKIP_SIP for the QgsPointXY constructor? 

It's a shame it isn't working, we have no way of building a QgsLineString with QgsPointXY elements on python.

File: python/plugins/processing/algs/qgis/Grid.py [hunk 1]
```diff
@@ -119,24 +123,24 @@ def processAlgorithm(self, parameters, context, feedback):
         crs = self.parameterAsCrs(parameters, self.CRS, context)
         bbox = self.parameterAsExtent(parameters, self.EXTENT, context, crs)
 
-        width = bbox.width()
-        height = bbox.height()
-        originX = bbox.xMinimum()
-        originY = bbox.yMaximum()
+        #width = bbox.width()
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/Grid.py]: remove?

File: python/plugins/processing/algs/qgis/Grid.py [hunk 2]
```diff
@@ -147,27 +151,134 @@ def processAlgorithm(self, parameters, context, feedback):
...
+                feat.setGeometry(QgsGeometry.fromPointXY(QgsPointXY(x,y)))
+                feat.setAttributes([x, y, x + hSpacing, y + vSpacing, id])
+                sink.addFeature(feat, QgsFeatureSink.FastInsert)
+
+                id += 1
+                count += 1
+                if int(math.fmod(count, count_update)) == 0:
+                    feedback.setProgress(int(count / cells * 100))
+
+    def _lineGrid(self, sink, bbox, hSpacing, vSpacing, hOverlay, vOverlay, feedback):
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/Grid.py]: i'm guessing this is directly copied from the removed alg?

**Review by nyalldawson — COMMENTED** [review]
I think there's some tests you'll need to update to reflect the new name too

File: python/plugins/processing/algs/qgis/Grid.py (hunk 1 above)
Comment (nirvn) [reply python/plugins/processing/algs/qgis/Grid.py]: Oups.

File: python/plugins/processing/algs/qgis/Grid.py (hunk 2 above)
Comment (nirvn) [reply python/plugins/processing/algs/qgis/Grid.py]: Yes, no change applied to the function within that def, straight forward copy/paste.

**Comment by DelazJ** [comment]
@nirvn the right tag for doc is [needs-docs] (double S for memo) otherwise your PR is not reported.

**Comment by nirvn** [comment]
@DelazJ , oh, I must have done that wrong for quite a few times already then. Thanks for the note.

**Pull request merged**